### ISSUE TITLE
Using discriminatedUnion with nativeEnum

### ISSUE THREAD

**Issue by jonasduever** [colinhacks/zod#3343]
There is an issue when using `discriminatedUnion` in combination with `nativeEnum`s. (`enum`s work fine.)

I created a playground to reproduce this: https://stackblitz.com/edit/zod-discriminated-union-enums?file=schemas%2FschemaTests.ts

The native enums look like this:
```ts
enum Type1 {
  FirstType = 'firstType',
}
enum Type2 {
  SecondType = 'secondType',
  ThirdType = 'secondType',
}
```
The error message when using them to parse an object with `type: 'firstType'` is:
```
Invalid discriminator value. Expected 'FirstType' | 'SecondType' | 'ThirdType'
```

zod seems to use the enum **keys** instead of the values here, which is wrong from my perspective. When using a `nativeEnum` in simpler schemas, it takes the value (`firstType` etc.) as expected.

As a workaround, we now use `enum([Type1.FirstType])` etc., but that is not really a clean solution IMO.

Can you confirm this is a bug in zod?

---
<details><summary>Playground Code</summary>
<p>

```typescript
import { z } from 'zod';

const testData = { type: 'firstType', data: { something: 'test' } };

/**
 * NON-NATIVE ENUMS
 */

const schema1 = z
  .object({
    type: z.enum(['firstType']),
    data: z.object({ something: z.string() }),
  })
  .strict();

const schema2 = z
  .object({
    type: z.enum(['secondType', 'thirdType']),
    data: z.object({ somethingElse: z.string() }),
  })
  .strict();

export const schemaUnion = z.discriminatedUnion('type', [schema1, schema2]);

const validatedTestData = schemaUnion.safeParse(testData);

console.info('Validated non-native enums (success)', validatedTestData);

/**
 * NATIVE ENUMS
 *
 * It seems to take the enum key (left side) instead of the value (right side).
 */

enum Type1 {
  FirstType = 'firstType',
}
enum Type2 {
  SecondType = 'secondType',
  ThirdType = 'secondType',
}

const schema1Native = z
  .object({
    type: z.nativeEnum(Type1),
    data: z.object({ something: z.string() }),
  })
  .strict();

const schema2Native = z
  .object({
    type: z.nativeEnum(Type2),
    data: z.object({ somethingElse: z.string() }),
  })
  .strict();

export const schemaUnionNative = z.discriminatedUnion('type', [
  schema1Native,
  schema2Native,
]);

const validatedTestDataNative = schemaUnionNative.safeParse(testData);

console.info('Validated native enums (fails)', validatedTestDataNative);

``` 

</p>
</details>

**Comment by colinhacks** [comment]
It is a bug. I have a fix here: https://github.com/colinhacks/zod/pull/3346

**Comment by colinhacks** [comment]
This fix has landed in Zod 3.23
